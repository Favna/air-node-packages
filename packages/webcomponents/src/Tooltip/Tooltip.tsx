import Box from '@material-ui/core/Box';
import MUITooltip, { TooltipProps as MUITooltipProps } from '@material-ui/core/Tooltip';
import classnames from 'classnames';
import React, { FC } from 'react';
import css from './Tooltip.scss';
import { customCss, dataQa } from 'typings';

export type TooltipProps = MUITooltipProps & {
  /** Data-qa tag to apply to the search bar and input element */
  'data-qa'?: dataQa;
  /** Custom CSS classes to pass to the button */
  customclasses?: customCss;
};

const Tooltip: FC<TooltipProps> = props => (
  <MUITooltip
    title={
      <Box data-qa={props['data-qa']}>
        {props.title}
        <span className={css.arrow} />
      </Box>
    }
    placement={props.placement}
    enterDelay={props.enterDelay || 300}
    classes={{ popper: css.tooltip, tooltip: classnames(css.tooltip, props.customclasses) }}
    PopperProps={{popperOptions: {modifiers: {arrow: {enabled: true}}}}}
  >
    {props.children}
  </MUITooltip>
);

export default Tooltip;