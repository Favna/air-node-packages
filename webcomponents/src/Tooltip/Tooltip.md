### Usage of the ToolTip component:

**Examples:**

```tsx
import React from 'react';

import IconButton from '@material-ui/core/IconButton';
import InfoIcon from '@material-ui/icons/Info';
import { Tooltip } from '@rws-air/webcomponents';

<Tooltip title='Tooltip Title' placement='right' enterDelay={50}>
  <IconButton tabIndex={-1} href=''>
    <InfoIcon />
  </IconButton>
</Tooltip>
```

```jsx
import React from 'react';

import IconButton from '@material-ui/core/IconButton';
import InfoIcon from '@material-ui/icons/Info';
import { Tooltip } from '@rws-air/webcomponents';

<Tooltip title='Tooltip Title' placement='right' enterDelay={50}>
  <IconButton tabIndex={-1} href=''>
    <InfoIcon />
  </IconButton>
</Tooltip>
```