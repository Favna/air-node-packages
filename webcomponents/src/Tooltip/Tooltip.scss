@import '../styles/vars';

@mixin tooltipDefaults {
  background-color: $white;
  border: 1px solid $light-grey;
  border-radius: 0;
  color: $text-black;
  font-size: 0.9rem;
}

/*
  Based on CSS Arrows from: http://www.cssarrowplease.com/
  $color is the color of the arrow
  $arrowSize is the size of the arrow
  $borderSize affects the size of the entire tooltip
              and should be smaller than $arrowSize
  $[top|bottom|left|right]Offset defines the offset the arrow has to the
              tooltip. 50% puts it in the middle of the tooltip of the given side
*/
@mixin popperArrowUpGenerator ($color, $arrowSize, $borderSize, $leftOffset) {
  &::after,
  &::before {
    border: solid transparent;
    bottom: 100%;
    content: ' ';
    height: 0;
    left: $leftOffset;
    pointer-events: none;
    position: absolute;
    width: 0;
  }

  &::after {
    border-bottom-color: $white;
    border-width: $arrowSize;
    margin-left: -$arrowSize;
  }

  &::before {
    border-bottom-color: $color;
    border-width: ($arrowSize + ($borderSize * $arrowModifier));
    margin-left: -($arrowSize + ($borderSize * $arrowModifier));
  }
}

@mixin popperArrowLeftGenerator ($color, $arrowSize, $borderSize, $topOffset) {
  &::after,
  &::before {
    border: solid transparent;
    content: ' ';
    height: 0;
    pointer-events: none;
    position: absolute;
    right: 100%;
    top: $topOffset;
    width: 0;
  }

  &::after {
    border-right-color: $white;
    border-width: $arrowSize;
    margin-top: -$arrowSize;
  }

  &::before {
    border-right-color: $color;
    border-width: ($arrowSize + ($borderSize * $arrowModifier));
    margin-top: -($arrowSize + ($borderSize * $arrowModifier));
  }
}

.tooltip[x-placement*='right'] > div {
  @include popperArrowLeftGenerator($light-grey, 10px, 1px, 25%);
  @include tooltipDefaults;
  font-size: 0.75rem;
  margin-left: 0.7rem;
  margin-top: 3.3rem;
  position: relative;
  width: 60%;
}

.tooltip[x-placement*='bottom'] > div {
  @include popperArrowUpGenerator($light-grey, 10px, 1px, 25%);
  @include tooltipDefaults;
  margin-top: 0.5rem;
  position: relative;
}
