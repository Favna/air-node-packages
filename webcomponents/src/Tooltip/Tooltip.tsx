import MUITooltip, { TooltipProps as MUITooltipProps } from '@material-ui/core/Tooltip';
import classnames from 'classnames';
import React, { FC, Fragment } from 'react';
import css from './Tooltip.scss';
import { customCss, DataQa } from 'typings';

export type TooltipProps = MUITooltipProps & {
  /** data-qa tag to apply to the search bar and input element */
  'data-qa'?: DataQa;
  /** Custom CSS classes to pass to the button */
  customclasses?: customCss;
};

const Tooltip: FC<TooltipProps> = props => (
  <MUITooltip
    data-qa={props['data-qa']}
    title={
      <Fragment>
        {props.title}
        <span className={css.arrow} />
      </Fragment>
    }
    placement={props.placement}
    enterDelay={props.enterDelay || 300}
    classes={{ popper: css.tooltip, tooltip: classnames(css.tooltip, props.customclasses) }}
    PopperProps={{
      popperOptions: {
        modifiers: {
          arrow: {
            enabled: true,
          },
        },
      },
    }}
  >
    {props.children}
  </MUITooltip>
);

export default Tooltip;