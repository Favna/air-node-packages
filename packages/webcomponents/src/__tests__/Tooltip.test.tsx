import { shallow } from 'enzyme';
import React from 'react';
import Tooltip from '../Tooltip/Tooltip';

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const simpleTooltip = shallow(<Tooltip title='SnapTitle'><span>Snappy Text</span></Tooltip>);
    expect(simpleTooltip).toMatchSnapshot();
  });

  test('Optional Props', () => {
    const simpleTooltip = shallow(
      <Tooltip title='SnapTitle' data-qa='snap-qa' style={{ borderLeftColor: 'red' }}>
        <span>Snappy Text</span>
      </Tooltip>
    );
    expect(simpleTooltip).toMatchSnapshot();
  });
});