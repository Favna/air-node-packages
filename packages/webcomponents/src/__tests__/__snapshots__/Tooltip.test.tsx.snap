// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Optional Props 1`] = `
<WithStyles(ForwardRef(Tooltip))
  classes={
    Object {
      "popper": undefined,
      "tooltip": "",
    }
  }
  enterDelay={300}
  placement="top"
  style={
    Object {
      "borderLeftColor": "red",
    }
  }
  title={
    <ForwardRef(StyledComponent)
      data-qa="snap-qa"
    >
      SnapTitle
    </ForwardRef(StyledComponent)>
  }
>
  <span>
    Snappy Text
  </span>
</WithStyles(ForwardRef(Tooltip))>
`;

exports[`Snapshot Testing Required Props 1`] = `
<WithStyles(ForwardRef(Tooltip))
  classes={
    Object {
      "popper": undefined,
      "tooltip": "",
    }
  }
  enterDelay={300}
  placement="top"
  title={
    <ForwardRef(StyledComponent)>
      SnapTitle
    </ForwardRef(StyledComponent)>
  }
>
  <span>
    Snappy Text
  </span>
</WithStyles(ForwardRef(Tooltip))>
`;
